test/unit/resource_manager/test_resource_manager.py

```diff
@@ -0,0 +1,96 @@
+import base64
+from pathlib import Path
+
+import pytest
+
+from voicevox_engine.resource_manager import ResourceManager, ResourceManagerError
+
+with_filemap_dir = Path(__file__).parent / "with_filemap"
+without_filemap_dir = Path(__file__).parent / "without_filemap"
+
+
+def b64encode_str(s: bytes) -> str:
+    return base64.b64encode(s).decode("utf-8")
+
+
+def _assert_resource(manager: ResourceManager, input_path: Path) -> None:
+    """
+    `input_path`で指定したファイルから正しくbase64が取得できるか確認する
+    また、ハッシュを取得し、対応するファイルから同じバイト列が取得できるか確認する
+    """
+    true_bytes = input_path.read_bytes()
+
+    assert manager.resource_str(input_path, "base64") == b64encode_str(true_bytes)
+
+    result_filehash = manager.resource_str(input_path, "hash")
+    result_path = manager.resource_path(result_filehash)
+    assert result_path.read_bytes() == true_bytes
+
+
+def test_with_filemap() -> None:
+    """
+    "filemap.json"があるディレクトリでのテスト
+    fimemapの生成方法
+    `python tools/generate_filemap.py --target_dir test/unit/resource_manager/with_filemap`
+    """
+    manager = ResourceManager(False)
+    manager.register_dir(with_filemap_dir)
+
+    png_path = with_filemap_dir / "dummy.png"
+    _assert_resource(manager, png_path)
+
+    wav_path = with_filemap_dir / "dummy.wav"
+    _assert_resource(manager, wav_path)
+
+    # 同じバイナリがある場合のテスト
+    same_wav_path = with_filemap_dir / "dummy_same_binary.wav"
+    assert wav_path.read_bytes() == same_wav_path.read_bytes()
+    _assert_resource(manager, same_wav_path)
+
+    # filemap.jsonに含まれないものはエラー
+    # NOTE: テキストは通常エンコードせずにJSONに含めるためResourceManagerでは管理しない
```

Comment:
filemap.jsonに含めるのかなと勘違いしそうなので、まあ言及なくて良さそう？
```suggestion
    # NOTE: 通常、テキストはResourceManagerで管理しない
```